### DISCUSSION TITLE
Ray Tracing Module - CRC fails for non-zero speed values

### DISCUSSION THREAD

**Discussion by neilb22** [NVlabs/sionna#241] [Q&A] [answered]
Hello Everyone.

I'm going through the Sionna tutorial on [Ray Tracing](https://nvlabs.github.io/sionna/examples/Sionna_Ray_Tracing_Introduction.html#GPU-Configuration-and-Imports). I've followed the tutorial example up until the line:

`a, tau = paths.cir()`

I've used `sionna.channel.cir_to_ofdm_channel` and `sionna.channel.cir_to_time_channel`. For encoding and decoding, I've used the Transport block encoder and decoders. 

The point where I am confused is wrt the lines here:

```
l_min, l_max = sionna.channel.time_lag_discrete_time_channel(RESOURCE_GRID.bandwidth)
l_tot = l_max- l_min + 1

# Apply Doppler shifts
paths.apply_doppler(sampling_frequency=15e3, 
                        num_time_steps=RESOURCE_GRID.num_time_samples+l_tot-1, # Number of OFDM symbols
                        tx_velocities=[0, 0, 0], # We can set additional tx speeds
                        rx_velocities=[3.0, 3.0, 0]) # Or rx speeds       
```

I get a CRC fail if I specify non-zero speed values for RX. The CRC passes if I specify zero speeds for `rx_velocities`. Is there a reason why this happens?

For comparison, I tried with Sionna's TDL model instead of the "paths.cir()" for RT, and in that case, I would get a CRC pass even with non-zero min and max speeds.

**Comment by neilb22** [comment]
Hi everyone. 

It's been a few weeks since I made this post. I have not been able to resolve the issue. May I request for assistance/suggestions regarding this?

Kind Regards,
Neil

**Comment by faycalaa** [comment]
Hi,

Could you share a minimal code snippet that reproduces the issue ?

**Comment by neilb22** [comment]
Hi Fayçal. 
PFA the notebook file. 
 rt_crc_issue.zip (Edited to remove file.)

The issue I am having is with the `paths.apply_doppler()` function, specifically with `rx_velocities`.

Kind Regards,
Neil

**Comment by faycalaa** [comment, marked as answer]
Hi Neil,

In the code you shared, when invoking the ``apply_doppler`` function, the sampling frequency should match the bandwidth and not the subcarrier spacing.

Also, please note that there is currently a bug in `apply_doppler()` (see #222).

**Comment by neilb22** [comment]
I see. Thank you very much for pointing out my mistake.  Much appreciated. The program works now as expected. Thank you for also mentioning the issue 222. I will keep that in mind.

Kind Regards, 
Neil
